### PR TITLE
fix: Namespace auto-creation

### PR DESCRIPTION
**Pull request by mayzhang2000**

### CODE REVIEW

**Comment by james-callahan** [comment]
Is it worth adding the option for labels on the created namespace? (to allow tracking which application/team a namespace belongs to?)

**Comment by mayzhang2000** [comment]
Thank you @james-callahan for your review! The intent of this PR is creating a basic namespace. We can create a different issue for the need of having the option for labels for namespace.

**Comment by mayzhang2000** [comment]
> Is it worth adding the option for labels on the created namespace? (to allow tracking which application/team a namespace belongs to?)

Thank you @james-callahan for your review! The intent of this PR is creating a basic namespace. We can create a different issue for the need of having the option for labels for namespace.

File: pkg/sync/sync_context.go [hunk 1]
```diff
@@ -564,9 +564,27 @@ func (sc *syncContext) getSyncTasks() (_ syncTasks, successful bool) {
 
 	sort.Sort(tasks)
 
+	if sc.createNamespace && sc.namespace != "" {
```
Comment (alexmt) [inline pkg/sync/sync_context.go:567]: I don't think this is required. I've checked manually and pre-sync state Namespace is executed first already.

**Pull request merged**